tests/instances.tcl

```diff
@@ -85,25 +84,23 @@ proc spawn_instance {type base_port count {conf {}} {base_conf_file ""}} {
             set cfg [open $cfgfile w]
         }
 
+        puts $cfg [format "tls-cert-file %s/../../tls/server.crt" [pwd]]
+        puts $cfg [format "tls-key-file %s/../../tls/server.key" [pwd]]
+        puts $cfg [format "tls-client-cert-file %s/../../tls/client.crt" [pwd]]
+        puts $cfg [format "tls-client-key-file %s/../../tls/client.key" [pwd]]
+        puts $cfg [format "tls-dh-params-file %s/../../tls/redis.dh" [pwd]]
+        puts $cfg [format "tls-ca-cert-file %s/../../tls/ca.crt" [pwd]]
         if {$::tls} {
             if {$::tls_module} {
                 puts $cfg [format "loadmodule %s/../../../src/redis-tls.so" [pwd]]
             }
-
-            puts $cfg "tls-port $port"
             puts $cfg "tls-replication yes"
             puts $cfg "tls-cluster yes"
-            # plaintext port, only used by plaintext clients in a TLS cluster
-            set pport [find_available_port $base_port $::redis_port_count]
             puts $cfg "port $pport"
-            puts $cfg [format "tls-cert-file %s/../../tls/server.crt" [pwd]]
-            puts $cfg [format "tls-key-file %s/../../tls/server.key" [pwd]]
-            puts $cfg [format "tls-client-cert-file %s/../../tls/client.crt" [pwd]]
-            puts $cfg [format "tls-client-key-file %s/../../tls/client.key" [pwd]]
-            puts $cfg [format "tls-dh-params-file %s/../../tls/redis.dh" [pwd]]
-            puts $cfg [format "tls-ca-cert-file %s/../../tls/ca.crt" [pwd]]
+            puts $cfg "tls-port $port"
         } else {
             puts $cfg "port $port"
+            puts $cfg "tls-port $pport"
```

Comment:
Our default build does tests with BUILD_TLS=yes, even though it doesn't run it with TLS, so I'm not sure this would work if you didn't build with TLS since I think it would fail to configure. We might need a dedicated TLS test that tests specifically with cluster-tls disabled.

Reply:
Yes, if we don't build with TLS but provide a "tls-port" config, the server wouldn't start properly.

I think we already have some TLS tests with `cluster-tls` disabled (execute "runtest-cluster" without "--tls").

Reply:
Oh... now I understand.
Yes you are right, we should not include any TLS stuff when executing "runtest-cluster" without "--tls".

> We might need a dedicated TLS test that tests specifically with cluster-tls disabled.

Let me try to rewrite the test.